onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -231,6 +231,8 @@ void ModelBuilder::RegisterInitializers() {
       shape.push_back(SafeInt<uint32_t>(dim));
     }
 
+    ORT_ENFORCE(!shape.empty(), "NNAPI does not support scalar initializer");
+
```

Comment:
Not quite understanding this. If it supports a rank 1 initializer with one element, can't it support a scalar? Both involve one element.

Reply:
NNAPI has very strict difference between scalar and and tensor, most of the data input will be tensor, while for ONNX, a scalar can be a tensor with rank 1 shape with 1 element or a tensor with empty shape, before we took a hack to add {1} shape to a scalar input, but right now this will be easily confused with a dynamic shape which has an empty shape, that's why I want to disable this scenario for now since this is not a very common case.


---
In reply to: [462090615](https://github.com/microsoft/onnxruntime/pull/4650#discussion_r462090615) [](ancestors = 462090615)